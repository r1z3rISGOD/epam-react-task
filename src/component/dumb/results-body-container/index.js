import {ResultsBodyContainer} from "./results-body-container"
export {ResultsBodyContainer}
