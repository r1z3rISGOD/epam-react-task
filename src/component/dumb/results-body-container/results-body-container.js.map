{"version":3,"file":"results-body-container.js","sourceRoot":"","sources":["results-body-container.tsx"],"names":[],"mappings":";;;AAAA,6BAA8B;AAC9B,+BAAiC;AACjC,gDAA6C;AAC7C,yCAAsC;AACtC,gDAA6C;AAC7C,sCAAuC;AAGvC,IAAM,yBAAyB,GAAG;IAChC,WAAW,EAAE,SAAS,CAAC,KAAK;IAC5B,MAAM,EAAE,SAAS,CAAC,MAAM;IACxB,aAAa,EAAE,SAAS,CAAC,MAAM;IAC/B,gBAAgB,EAAE,SAAS,CAAC,IAAI;IAChC,YAAY,EAAE,SAAS,CAAC,IAAI;CAC7B,CAAA;AAEM,IAAM,oBAAoB,GAA4D,UAAC,EAAsE;QAApE,WAAW,iBAAA,EAAE,MAAM,YAAA,EAAE,aAAa,mBAAA,EAAE,gBAAgB,sBAAA,EAAE,YAAY,kBAAA;IAChK,iBAAS,CAAC;QACR,gBAAgB,CAAC,WAAW,EAAE,aAAa,EAAE,MAAM,CAAC,CAAA;IACtD,CAAC,CAAC,CAAA;IAEF,IAAI,WAAW,CAAC,MAAM,KAAK,CAAC,EAAE;QAC5B,OAAO,CACD,6BAAK,SAAS,EAAC,wBAAwB;YACnC,oBAAC,0BAAW;gBACR,6BAAK,SAAS,EAAC,wBAAwB;oBACnC,8BAAM,SAAS,EAAC,oBAAoB,uBAAwB,CAC1D,CACI,CACZ,CACX,CAAA;KACF;IACD,OAAO,CACG,6BAAK,SAAS,EAAC,wBAAwB;QACnC,oBAAC,0BAAW;YACR,6BAAK,SAAS,EAAC,qBAAqB,IAC/B,WAAW,CAAC,GAAG,CAAC,UAAA,IAAI;gBACjB,OAAA,oBAAC,0BAAW,IACR,MAAM,EAAE,MAAM,EACd,KAAK,EAAE,IAAI,CAAC,KAAK,EACjB,GAAG,EAAE,IAAI,CAAC,EAAE,EACZ,IAAI,EAAE,IAAI,CAAC,YAAY,EACvB,OAAO,EAAE,IAAI,CAAC,WAAW,EACzB,MAAM,EAAE,IAAI,CAAC,MAAM,EACnB,EAAE,EAAE,IAAI,CAAC,EAAE,EACX,YAAY,EAAE,YAAY,GAC5B;YATF,CASE,CACL,CACC,CACI,CACZ,CACf,CAAA;AACH,CAAC,CAAA;AApCY,QAAA,oBAAoB,wBAoChC"}