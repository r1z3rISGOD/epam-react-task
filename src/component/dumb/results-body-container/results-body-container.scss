@import "./src/_variables/variables.scss";

.results-body {
  width: 85%;
  height: 100%;
  margin: auto;
}

.results-body__text {
  font-family: $font;
  font-size: 5rem;
}

.results-body__films {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
  margin-top: 50px;
}

.results-body__no-films {
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
}

.results-body-container {
  background-color: #fcfffd;
  min-height: 100vh;
}

.results-body__img {
  width: 100px;
  height: 200px;
}

.results-body__item {
  width: 100px;
  height: 200px;
}
