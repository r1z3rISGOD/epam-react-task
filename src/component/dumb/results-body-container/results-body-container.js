"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResultsBodyContainer = void 0;
var React = require("react");
var react_1 = require("react");
var results_body_1 = require("../results-body");
require("./results-body-container.scss");
var results_item_1 = require("../results-item");
var PropTypes = require("prop-types");
var resultsBodyContainerProps = {
    fetchedData: PropTypes.array,
    active: PropTypes.string,
    searchSetting: PropTypes.string,
    fetchingAllFilms: PropTypes.func,
    fetchingFilm: PropTypes.func
};
var ResultsBodyContainer = function (_a) {
    var fetchedData = _a.fetchedData, active = _a.active, searchSetting = _a.searchSetting, fetchingAllFilms = _a.fetchingAllFilms, fetchingFilm = _a.fetchingFilm;
    react_1.useEffect(function () {
        fetchingAllFilms(fetchedData, searchSetting, active);
    });
    if (fetchedData.length === 0) {
        return (React.createElement("div", { className: "results-body-container" },
            React.createElement(results_body_1.ResultsBody, null,
                React.createElement("div", { className: "results-body__no-films" },
                    React.createElement("span", { className: "results-body__text" }, " No films found ")))));
    }
    return (React.createElement("div", { className: "results-body-container" },
        React.createElement(results_body_1.ResultsBody, null,
            React.createElement("div", { className: "results-body__films" }, fetchedData.map(function (film) {
                return React.createElement(results_item_1.ResultsItem, { active: active, title: film.title, key: film.id, year: film.release_date, picture: film.poster_path, genres: film.genres, id: film.id, fetchingFilm: fetchingFilm });
            })))));
};
exports.ResultsBodyContainer = ResultsBodyContainer;
//# sourceMappingURL=results-body-container.js.map