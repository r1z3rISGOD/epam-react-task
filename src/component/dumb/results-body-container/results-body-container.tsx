import * as React from 'react'
import { useEffect } from 'react'
import { ResultsBody } from '../results-body'
import './results-body-container.scss'
import { ResultsItem } from '../results-item'
import { fetchData } from '../../services/fetchService'
import { useLocation } from 'react-router'
import { useDispatch } from 'react-redux'
import { changeSearchRequestedData, putFetchedArrayToStore } from '../../../store/actions'

interface ResultsBodyContainerProps {
    fetchedData: any[]
    active: string
    searchSetting: string
}

export const ResultsBodyContainer : React.FC<ResultsBodyContainerProps> = ({ fetchedData, active, searchSetting }) => {
  const dispatch = useDispatch()
  const location = useLocation().search

  useEffect(() => {
    const fetching = async () => {
      const params = new URLSearchParams(location)
      const searchRequest = params.getAll('request')[0]
      if (fetchedData.length === 0 && searchRequest) {
        const newFetchedFilms = await fetchData(searchRequest, searchSetting, active)
        if (newFetchedFilms.data.length > 0) {
          dispatch(putFetchedArrayToStore(newFetchedFilms.data))
          dispatch(changeSearchRequestedData(searchRequest))
        }
      }
    }
    fetching()
  })

  if (fetchedData.length === 0) {
    return (
          <div className="results-body-container">
              <ResultsBody>
                  <div className="results-body__no-films">
                      <span className="results-body__text"> No films found </span>
                  </div>
              </ResultsBody>
          </div>
    )
  }
  return (
            <div className="results-body-container">
                <ResultsBody>
                    <div className="results-body__films">
                        {fetchedData.map(film =>
                            <ResultsItem
                                active={active}
                                title={film.title}
                                key={film.id}
                                year={film.release_date}
                                picture={film.poster_path}
                                genres={film.genres}
                                id={film.id}
                            />
                        )}
                    </div>
                </ResultsBody>
            </div>
  )
}
