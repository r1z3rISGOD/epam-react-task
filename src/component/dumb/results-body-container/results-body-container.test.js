import { ResultsBodyContainer } from './results-body-container.tsx'
import React from 'react'

const film = [
  {
    id: 412302,
    title: 'Gemini',
    tagline: '',
    vote_average: 10,
    vote_count: 3,
    release_date: '2018-03-30',
    poster_path: 'https://image.tmdb.org/t/p/w500/oIltQs7MPk7VQFG3DJfgC63mShU.jpg',
    overview: 'A heinous crime tests the complex relationship between a tenacious personal assistant and her Hollywood starlet boss. As the assistant travels across Los Angeles to unravel the mystery, she must stay one step ahead of a determined policeman and confront her own understanding of friendship, truth and celebrity.',
    budget: 0,
    revenue: 0,
    genres: [
      'Mystery',
      'Thriller'
    ],
    runtime: 92
  }
]

const mockFetchingAllFilms = jest.fn((fetchedData, searchSetting, active) => console.log(fetchedData, searchSetting, active))

describe('results-body-container tests', () => {
  it('results body container test 1', () => {
    const comp = shallow(<ResultsBodyContainer fetchedData={[]} fetchingAllFilms={mockFetchingAllFilms} searchSetting='title' active='vote_average'/>)
    expect(comp).toMatchSnapshot()
  })
  it('results body container test 2', () => {
    const comp = shallow(<ResultsBodyContainer fetchedData={film} fetchingAllFilms={mockFetchingAllFilms} searchSetting='title' active='vote_average'/>)
    expect(comp).toMatchSnapshot()
  })
})
