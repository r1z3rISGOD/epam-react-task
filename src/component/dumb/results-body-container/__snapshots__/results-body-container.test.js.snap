// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`results-body-container tests results body container test 1 1`] = `
<div
  className="results-body-container"
>
  <ResultsBody>
    <div
      className="results-body__no-films"
    >
      <span
        className="results-body__text"
      >
         No films found 
      </span>
    </div>
  </ResultsBody>
</div>
`;

exports[`results-body-container tests results body container test 2 1`] = `
<div
  className="results-body-container"
>
  <ResultsBody>
    <div
      className="results-body__films"
    >
      <ResultsItem
        active="vote_average"
        genres={
          Array [
            "Mystery",
            "Thriller",
          ]
        }
        id={412302}
        key="412302"
        picture="https://image.tmdb.org/t/p/w500/oIltQs7MPk7VQFG3DJfgC63mShU.jpg"
        title="Gemini"
        year="2018-03-30"
      />
    </div>
  </ResultsBody>
</div>
`;
